src/helpers/exceptions.py

```diff
@@ -0,0 +1,28 @@
+# Snowplow fetching
+class EventObjectFetchError(Exception):
```

Comment:
🔧 Happy to discuss, but upon further thinking, I think we should *not* have custom exceptions here, and we should just let any boto3 exceptions surface to the top. They have [a lot of exceptions that are detailed](https://github.com/boto/botocore/blob/develop/botocore/exceptions.py) and would provide more specific reasons why something didn't work. For example, using the `object_summary.get()` method might error out in a few ways. We would want to know if there was an HTTP timeout, or if there was a permission issue, or something else.

So my proposed change would be to *not* wrap boto things in `try/except` and let the boto exceptions bubble to the top, unless we want to handle them for retrying purposes *or* bubbling the exception up with added information.

Reply:
Wow, that link's a great find! Honestly, I'm all for deleting the try-except blocks, even the ones for decompressing and parsing. We can deal with exceptions when they happen.

Reply:
Done!